lib/package_source.py

```diff
@@ -19,20 +21,32 @@ def _hg_download(source, directory):
     dest = os.path.join(directory, repo_name)
     command = 'hg '
 
-    if proxy:
-        command += '-c http_proxy.host="{}" '.format(proxy)
-
     commit_id = hg_source.get('commit_id')
     branch = hg_source.get('branch')
 
     if commit_id is None and branch is None:
         raise ValueError('invalid hg source dict: missing both `commit_id` '
                          'and `branch`')
 
-    command += 'clone "{}" -r "{}" "{}"'.format(hg_source['src'],
+    command += 'clone "{}" -r "{}" "{}" '.format(hg_source['src'],
                                                 hg_source['branch'],
                                                 dest)
-    utils.run_command(command)
+    if proxy:
+        command += '--config http_proxy.host="{}" '.format(proxy)
+    command += "--ssh '/usr/bin/env ssh -o ConnectTimeout={}'"
+
+    def _clone_repository(timeout):
+        cmd = command.format(timeout)
+        utils.run_command(cmd)
+
+    def _is_timeout_error(e):
```

Comment:
I like`e` better (because that's what I most frequently see in python code), but if `exc` is what you agreed upon I'll just go along with that.